src/App.js

```diff
@@ -1,26 +1,36 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Fragment, useState } from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import axios from 'axios';
+import Home from './components/pages/Home';
+import Courses from './components/pages/Courses';
+
+const App = () => {
+  const [courses, setCourses] = useState('');
+
+  // Get courses
+  const getCourses = async () => {
+    const res = await axios.get('http://localhost:5000/courses');
```

Comment:
Let's make a wrapper for `axios` called either `AppRequest.js` or `Requestor.js` and put it under `/components` folder.

What that wrapper will do is to auto prepend `http://localhost:5000` to any request that comes in from rest of the app. That way we can setup environment variable later for dev vs prod to swap out that prepending url to hit local vs actual server that is running.

Reply:
It should look something along the lines of:

(make sure to check syntax. I'm just writing below as an example snippet not as something that fully works)

```js
import axios from 'axios';

// This will use environment variable later
const serverUrl = 'http://localhost:5000';

// Name of class should reflect the file name, which ever you may choose
class Requestor {
  private serverUrl;

  constructor(serverUrl) {
    this.serverUrl = serverUrl;
  }

  get(url, config) {
    return axios.get(`${serverUrl}${url}`, config);
  }

  ...
}

const requestor = new Requestor(serverUrl);
export { requestor };
```

```js
const courses = await axios.get('/courses');
```

Reply:
Change `res` to `coursesResponse` to be more descriptive.

Reply:
Should I use class and constructor? not arrow functional component?

Reply:
You can just return functions too if you prefer that way,

```js
import axios from 'axios';
// import { SERVER_URL } from '/environments';
const SERVER_URL = 'http://localhost:5000';

export const requestor = {
  get(url, config) {
    if (!url || url[0] !== '/') {
      return Promise.reject('Url should be prefixed with slash');
    }
    return axios.get(`${SERVER_URL}${url}`, config);
  },
  ...
};
```